ecmascript/parser/src/parser/stmt/module_item.rs

```diff
@@ -177,18 +177,90 @@ impl<'a, I: Tokens> Parser<I> {
     }
 
     /// Parse `foo`, `foo2 as bar` in `import { foo, foo2 as bar }`
-    fn parse_import_specifier(&mut self) -> PResult<ImportSpecifier> {
+    fn parse_import_specifier(&mut self, is_type_only_import: bool) -> PResult<ImportSpecifier> {
         let start = cur_pos!(self);
         match cur!(self, false) {
             Ok(&Word(..)) => {
-                let orig_name = self.parse_ident_name()?;
+                let mut is_type_only = false;
+                let mut orig_name = self.parse_ident_name()?;
+
+                // Handle:
+                // `import { type xx } from 'mod'`
+                // `import { type xx as yy } from 'mod'`
+                // `import { type as } from 'mod'`
+                // `import { type as as } from 'mod'`
+                // `import { type as as as } from 'mod'`
+                if orig_name.sym == js_word!("type")
+                    && is!(self, IdentName)
+                    && self.syntax().typescript()
```

Comment:
Also, please move `self.syntax().typescript()` to first condition.

Reply:
The reason why I don't put it first is that most users won't use "type" as specifier then this short circuit can bring better performance.

Reply:
Is u64 comparison faster than bool check?

Also, `self.syntax().typescript() ` is used to remove code for typescript parser.